pkg/types/types.go

```diff
@@ -154,35 +154,78 @@ func (m *ComMatrix) WriteMatrixToFileByType(utilsHelpers utils.UtilsInterface, f
 	return nil
 }
 
-// GenerateMatrixDiff generates the diff between mat1 to mat2.
-func (m *ComMatrix) GenerateMatrixDiff(mat2 *ComMatrix) (string, error) {
-	colNames, err := getComMatrixHeadersByFormat(FormatCSV)
-	if err != nil {
-		return "", fmt.Errorf("error getting commatrix CSV tags: %v", err)
+// Combine generates a new sorted matrix from m and other with no duplications.
```

Comment:
I think if call it add, add the two together and do the clean (remove dups and sort) is will do the same with less changes?

Reply:
you mean adding the matrix at the end without removing the duplicates, and than call the clean function?

Reply:
As part of the generate you can add the two mats with a single append and call clean.

Reply:
okay yeah. Though maybe we can change the name of the method `CleanComDetails` to `SortAndRemoveDuplicates`, I think it is more clear that way. what do you think?

Reply:
Totally agree